File: Data Structures & Algorithms/top-k-elements-in-list/submission-4.py
import heapq

class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
         nums_freqs = {}
         for n in nums:
            nums_freqs[n] = nums_freqs.get(n ,0) + 1

         heap = []
         for n, f in nums_freqs.items():
            if len(heap) < k:
               heapq.heappush(heap,(f,n))
            else:
               heapq.heappushpop(heap,(f,n))

         freq_buckets = heap
         result = []
         for i in range(len(freq_buckets)): 
            result.append(freq_buckets[i][1])
         return result

File: Data Structures & Algorithms/missing-number/submission-4.rs
impl Solution {
    pub fn missing_number(nums: Vec<i32>) -> i32 {
        let mut x_r = 0;
        for i in 1..=nums.len(){
            x_r ^= i as i32; 
        }
        for n in nums {
            x_r ^= n;
        }
        x_r
    }
}

// 1
